src/viewer/scene/sectionCaps/SectionCaps.js

```diff
@@ -0,0 +1,675 @@
+import { math } from "../math/math.js";
+import { Mesh } from "../mesh/Mesh.js";
+import { ReadableGeometry } from "../geometry/ReadableGeometry.js";
+import { buildLineGeometry } from "../geometry/index.js";
+import { PhongMaterial } from "../materials/PhongMaterial.js";
+import earcut from '../libs/earcut.js';
+
+const epsilon = 1e-6;
+const worldUp    = [0, 1, 0];
+const worldRight = [1, 0, 0];
+const tempVec3a = math.vec3();
+const tempVec3b = math.vec3();
+const tempVec3c = math.vec3();
+const tempVec3d = math.vec3();
+
+function pointsEqual(p1, p2) {
+    return (
+        Math.abs(p1[0] - p2[0]) < epsilon &&
+        Math.abs(p1[1] - p2[1]) < epsilon &&
+        Math.abs(p1[2] - p2[2]) < epsilon
+    );
+}
+
+class SectionCaps {
```

Comment:
Need a class comment - maybe something similar to [`SAO`](https://github.com/xeokit/xeokit-sdk/blob/master/src/viewer/scene/postfx/SAO.js)